homeassistant/components/shelly/binary_sensor.py

```diff
@@ -50,6 +64,21 @@
     ("sensor", "vibration"): BlockAttributeDescription(
         name="Vibration", device_class=DEVICE_CLASS_VIBRATION
     ),
+    ("input", "input"): BlockAttributeDescription(
+        name="Input",
+        device_class=DEVICE_CLASS_POWER,
+        removal_condition=lambda s, b: is_momentary_input(s, b),
```

Comment:
It'd be great to also expose input events here (short, long pushes etc.).

I think this should do it:
```python
        device_state_attributes=lambda block: {
            "event": block.inputEvent,
        },
```

Reply:
Binary sensors are only created if the button settings is not momentary and in this case the device does not generate events (short, long, etc)
Inputs events will be added in the next PR, not fired from inside an entity, but from the device wrapper.
This was request by @balloob